persistence/docs/SAVEPOINTS_ROLLBACKS.md

```diff
@@ -0,0 +1,146 @@
+
+    > TODO: **Discuss what is appropriate to do here**
+
+- [**Savepoints**] The simplest version could be the database transaction that can simply be discarded, basically the uncommitted transaction sits in memory until it's flushed to storage and rolling back to a savepoint would be as simple as discarding the non-pristine version of the state in memory.
+
+- [**Rollbacks**] Rolling back to a savepoint would mean not only that the state has been restored to the previous savepoint but also that the node has to go back into a state that allows it to proceed with its normal operation (i.e. all the modules should behave as if nothing has happened)
+
+- [**Extensive testing**] We need to make sure that we have a good test coverage for all the above scenarios and that we can simulate failures in a controlled environment.
+
+### Improvements over MVP (Ideas)
+
+Apart from internal failures that should resolve themselves automatically whenever possible, nodes might require a way to save their state and restore it later, not necessarily at the previous block height. This could be useful for a number of reasons:
+
+- To allow nodes to recover from an unforeseen crash (bugs)
+- To facilitate socially coordinated rollbacks of the chain to a specific block height in the case of a consensus failure/chain-halt
+- To improve operator experience when managing and upgrading their fleet of nodes
+- And more...
+
+Performing operations like: copying datadirs, compressing them, etc. is probably not the best approach.
+Having a first-class support for savepoints and rollbacks would be, IMHO, a much better solution.
+
+A more advanced/flexible version of **Savepoints** could be to efficiently serialize the state into an artifact (non-trivial also because we have multiple data-stores but definitely possible), maybe leveraging previous savepoints to only store the changes that have been made since the last one and/or using some version of a WAL (Write-Ahead log that records the changes that happened).
+
+#### Savepoints
+
+A savepoint must have the following properties:
+
+- It must be able to be created at any block height
+- It must be self-contained and/or easy to move around (e.g. a file/archive)
+- The action of creating a savepoint must be atomic (i.e. it must be impossible to create a savepoint that is incomplete)
+- The action of creating a savepoint must be easy to perform (e.g. a CLI command and/or a flag being passed to the node binary when starting it)
+- The action of creating a savepoint must be as fast as possible
+- The operator should be informed with meaningful messages about the progress of the savepoint creation process (telemetry, logging and stdout come to mind)
+- It should be, as much as possible, compact
```

Comment:
What do you mean by compact?

Reply:
I mean packed together, dense, zipped with a good compression ratio since speed of dissemination (download/upload) is IMHO more important than the one-off speed of compression.

Should I clarify?